idea/src/org/jetbrains/kotlin/idea/inspections/branchedTransformations/IfThenToSafeAccessInspection.kt

```diff
@@ -83,6 +83,9 @@ class IfThenToSafeAccessInspection : AbstractApplicabilityBasedInspection<KtIfEx
         negatedClause == null && baseClause.isUsedAsExpression(context) -> false
         negatedClause != null && !negatedClause.isNullExpression() -> false
         else -> baseClause.evaluatesTo(receiverExpression) || baseClause.hasFirstReceiverOf(receiverExpression) ||
-                receiverExpression is KtThisExpression && getImplicitReceiver()?.let { it.type == receiverExpression.getType(context) } == true
+                receiverExpression is KtThisExpression && getImplicitReceiver()?.let { it.type == receiverExpression.getType(context) } == true ||
+                (baseClause as? KtCallExpression)?.valueArguments?.map { it.getArgumentExpression() }?.let {
+                    it.any { it?.evaluatesTo(receiverExpression) == true } && it.all { it is KtNameReferenceExpression }
```

Comment:
Nested `it`s are very hard to read here.

Reply:
And I'd extract the whole condition (beginning from `baseClause as?`) into a function and simplify it a bit, for example to avoid safe call chains and `let`